lib/index.js

```diff
@@ -653,10 +695,26 @@ MapboxGeocoder.prototype = {
       if (reverseGeocodeCoordRgx.test(trimmedSearch)) {
         search = search.replace(/,/g, ' ');
       }
+      if(this.options.version === 'v6' && config.mode) {
+        config.permanent = config.mode === 'mapbox.places-permanent';
+        delete config.mode;
+      }
```

Comment:
This should be moved outside of the switch block, as it could apply to forward or reverse.